apps/shell/tash_main.c

```diff
@@ -121,12 +121,12 @@ char *tash_read_input_line(int fd)
 	int nbytes;
 	int char_idx;
 	bool is_tab_pressed = false;
-#if TASH_MAX_STORE > 0
+	int prev_cmd_len = 0;
 	bool is_esc_pressed = false;
+	char direction;
+#if TASH_MAX_STORE > 0
```

Comment:
Please add a description to the commit (1-2 lines) explaining the idea behind making this change to solve the issue at hand.